### PR TITLE
Expose option to disable CRC-32 computation during `torch.save`

### PR DESCRIPTION
**Pull request by mikaylagawarecki**
Option only works in open source, not internal

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #137735

Differential Revision: [D64545647](https://our.internmc.facebook.com/intern/diff/D64545647)

### CODE REVIEW

File: torch/serialization.py [hunk 1]
```diff
@@ -167,6 +169,34 @@ def set_default_load_endianness(endianness):
...
+def get_default_crc32_options() -> bool:
+    """
+    Get whether :func:`torch.save` computes and writes crc32 for each record.
+
+    Defaults to ``True``.
+    """
+    return _compute_crc32
+
+
+def set_default_crc32_options(compute_crc32: bool):
```
Comment (albanD) [inline torch/serialization.py:184]: nit: why `set_default_crc32_options` vs `set_crc32_options` ? There is no way to override this flag from the end user of torch.save point of view.

File: third_party/miniz-2.1.0/miniz.h [hunk 2]
```diff
@@ -1001,7 +1001,8 @@ typedef enum {
     MZ_ZIP_FLAG_VALIDATE_HEADERS_ONLY = 0x2000,     /* validate the local headers, but don't decompress the entire file and check the crc32 */
     MZ_ZIP_FLAG_WRITE_ZIP64 = 0x4000,               /* always use the zip64 file format, instead of the original zip file format with automatic switch to zip64. Use as flags parameter with mz_zip_writer_init*_v2 */
     MZ_ZIP_FLAG_WRITE_ALLOW_READING = 0x8000,
-    MZ_ZIP_FLAG_ASCII_FILENAME = 0x10000
+    MZ_ZIP_FLAG_ASCII_FILENAME = 0x10000,
+    MZ_ZIP_FLAG_DO_NOT_COMPUTE_CRC32 = 0x20000, /* don't compute the crc32 of file data that's being added. */
```
Comment (albanD) [inline third_party/miniz-2.1.0/miniz.h:1005]: I forgot what's the story with these changes and if they will work internally?

**Review by albanD — COMMENTED** [review]
Only small nits, sounds pretty good in general!

File: third_party/miniz-2.1.0/miniz.h (hunk 2 above)
Comment (mikaylagawarecki) [reply third_party/miniz-2.1.0/miniz.h:1005]: won't work internally as internal uses a different version of miniz

File: torch/serialization.py (hunk 1 above)
Comment (mikaylagawarecki) [reply torch/serialization.py:184]: will use `set_crc32_options`

**Review by albanD — APPROVED** [review]
SGTM !

**Comment by clee2000** [comment]
@pytorchbot revert -m "failing internally D64438525, probably needs gating" -c ghfirst

```
serialize/inline_container.cc:701:14: error: use of undeclared identifier 'MZ_ZIP_FLAG_DO_NOT_COMPUTE_CRC32'
    flags |= MZ_ZIP_FLAG_DO_NOT_COMPUTE_CRC32;
```

**Comment by mikaylagawarecki** [comment]
@mikaylagawarecki has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D64545647).

**Comment by wdvr** [comment]
as discussed with @mikaylagawarecki and @izaitsevfb - this has been reimported as a different PR to circumvent internal export check failures. I will close this one unmerged if that one is merged in.

**Pull request closed without merging**